ocaml/testsuite/tests/typing-local/close_at_apply.ml

```diff
@@ -0,0 +1,44 @@
+(* TEST
+   flambda2;
+   {
+     ocamlopt_flags="-Oclassic";
+   }{
+     ocamlopt_flags="-O3";
+   }
+*)
+
+external ignore : ('a[@local_opt]) -> unit = "%ignore"
+external opaque_identity : ('a[@local_opt]) -> ('a[@local_opt]) = "%opaque"
+
+external ( +. )
+  :  (float[@local_opt])
+  -> (float[@local_opt])
+  -> (float[@local_opt])
+  = "%addfloat"
+
+let locally_allocate =
+  let f = 1.0 in
+  fun [@inline never] () -> ignore (opaque_identity (f +. 1.0) : float)
+;;
+
+let foo list =
+  (* The Simplif local function optimization will transform [run] and
+     [wrapper] into continuations. *)
```

Comment:
Please annotate such functions with `[@local]` if your test relies on the optimisation